client/my-sites/plans-features-main/components/utils/utils.ts

```diff
@@ -56,3 +58,42 @@ export const getHidePlanPropsBasedOnThemeType = ( themeType: string ) => {
 
 	return {};
 };
+
+/**
+ * Ordered array of interval types from shortest to longest term.
+ */
+const INTERVAL_TYPES_ORDERED: SupportedUrlFriendlyTermType[] = [
+	'monthly',
+	'yearly',
+	'2yearly',
+	'3yearly',
+];
+
+/**
+ * Ensures that the requested intervalType is compatible with the current plan's term.
+ * Users can only select interval types that are equal to or longer than their current plan's interval.
+ */
+export const ensureCompatibleIntervalType = (
```

Comment:
I think my main concern with these two entities (`INTERVAL_TYPES_ORDERED` and `ensureCompatibleIntervalType`) is that they feel disconnected from the core logic and potentially replicate a concept that already exists in the code (the `supportedIntervalType` value). So we have a "compatible interval" concept and a "supported interval" concept. How are these different? Is there a way to only have one and not the other? 

In the future, I am assuming that it will be hard to maintain both. More so when the list is hardcoded here as well. 🤷